packages/fdc3-agent-proxy/src/channels/DefaultChannelSupport.ts

```diff
@@ -56,31 +65,37 @@ export class DefaultChannelSupport implements ChannelSupport {
       payload: {},
     };
     const response = await this.messaging.exchange<GetCurrentChannelResponse>(request, 'getCurrentChannelResponse');
-
-    if (response.payload?.channel?.id) {
+    //handle successful responses - errors will already have been thrown by exchange above
+    if (response.payload.channel) {
```

Comment:
can't we use your new throwIfUndefined here?

Reply:
Have put it in and removed the log message, but the else has to stay so that we return on all branches.